plone/namedfile/scaling.py

```diff
@@ -509,16 +507,18 @@ def scale(
         if info is None:
             return  # 404
 
-        info["srcset"] = self.calculate_srcset(
-            fieldname=fieldname,
-            height=height,
-            width=width,
-            direction=direction,
-            scale=scale,
-            storage=storage,
-            **parameters
-        )
-        info["fieldname"] = fieldname
+        if "srcset" not in info:
```

Comment:
So the `srcset` is actually stored in the `ScalesDict`? I did not expect that. You are right though.

A downside of not updating it always, is that the srcset configuration may have changed. Maybe 3x was not enabled earlier and now it is. Or the image quality at 2x was increased. You do not see the improvements here then.
Could be a minor issue.

Is this the part that gives the most performance improvement? I expect that this is mostly because the scale gets generated. Or did you try and measure after all the scales, including 2x and 3x were already generated? 

It may be good to wait until my improvements from PR #113 are approved (maybe by you) and merged. Then measure again. There I make a split between creating a unique id for the scale and actually generating the scale.

Reply:
> Is this the part that gives the most performance improvement? I expect that this is mostly because the scale gets generated. Or did you try and measure after all the scales, including 2x and 3x were already generated?
>
> It may be good to wait until my improvements from PR #113 are approved (maybe by you) and merged. Then measure again. There I make a split between creating a unique id for the scale and actually generating the scale.

Our tests are all within all the scale generated. So probably a different path from your improvements.

I finally agree with you, we can wait until you have merged your current works, happy to review them, and then reevaluate this.